Eastern-Research-Group/mywaterway
# Feature/move survey mapping to json

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3411201 - This does not fully complete this ticket.

## Main Changes:
* Created a function for fetching lookup files from the hmw proxy/s3. 
* Created a new context for holding the lookup files.
* Converted all references to surveyMapping.js file into using the new custom hooks for fetching the surveyMapping.json file from the proxy/s3.
  * I did not encounter any React.useEffect situations when converting this file, so I went ahead and converted waterTypeOptions as well.
* Converted all references to waterTypeOptions.js file into using the new custom hooks for fetching the waterTypeOptions.json file from the proxy/s3. 

## Steps To Test:
1. Navigate to http://localhost:3000/state/KS/water-quality-overview
2. Open the chrome dev tools.
3. On the network tab type "proxy" into the filter.
4. Verify the "waterTypeOptions" and "surveyMapping" files were only fetched one time using the proxy.
5. Verify the site specific data is displaying.
6. Select "Rivers and Streams" from the "Water Type" drop down.
7. Verify the survey data is displaying.
8.  Block the "surveyMapping" fetch.
9. Reload the page
10. Verify the "surveyMapping" fetch is called 4 times and an error is displayed on the screen.
11. Unblock the "surveyMappng" fetch and block the "waterTypeOptions" fetch.
12. Reload the page
13. Verify the "waterTypeOptiosn" fetch is called 4 times and an error is displayed on the screen. 

## TODO:
- [x] Come up with a list of all of the lookup files we want to move to S3. We might need to have a meeting about this. 
- [ ] Convert over from using hard-coded lookup files to using proxy/s3. 

This [commit](https://github.com/Eastern-Research-Group/mywaterway/commit/c3fa83488b4c2218512603ed4290c925c29d2c9b) can be used as a reference on how to convert the hard coded lookup file into using the S3 lookup file.
